c7n/resources/asg.py

```diff
@@ -103,6 +98,12 @@ def initialize(self, asgs):
                     cfg['LaunchConfigurationName']: cfg for cfg in configs}
                 return
 
+        """Get launch configs for the set of asgs"""
+        config_names = set()
+        for a in asgs:
+            if 'LaunchConfigurationName' in a:
+                config_names.add(a['LaunchConfigurationName'])
```

Comment:
if the launch config name is the asg name, then ... shouldn't it be config_names.add(a.get('LaunchConfigurationName', a['AutoScalingGroupName']))